backend/app/usecase/repo/userurl_fake.go

```diff
@@ -13,6 +13,11 @@ func (u UserURLRelationFake) CreateRelation(user entity.User, url entity.URL) er
 	return nil
 }
 
+// FindAllURLsByUser fetches list of URLs associated with given user
+func (u UserURLRelationFake) FindAllURLsByUser(user entity.User) ([]entity.URL, error) {
```

Comment:
Please provide in memory implementation here. You will need it when testing search business logic.